document-parsers/langchain4j-document-parser-apache-pdfbox/src/main/java/dev/langchain4j/data/document/parser/apache/pdfbox/ApachePdfBoxDocumentParser.java

```diff
@@ -18,19 +20,25 @@ public class ApachePdfBoxDocumentParser implements DocumentParser {
 
     @Override
     public Document parse(InputStream inputStream) {
-        try {
-            PDDocument pdfDocument = PDDocument.load(inputStream);
+        try (PDDocument pdfDocument = PDDocument.load(inputStream)) {
             PDFTextStripper stripper = new PDFTextStripper();
             String text = stripper.getText(pdfDocument);
-            pdfDocument.close();
-
             if (isNullOrBlank(text)) {
                 throw new BlankDocumentException();
             }
-
-            return Document.from(text);
+            Metadata metadata = toMetadata(pdfDocument.getDocumentInformation());
+            return Document.from(text, metadata);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
+
+    private Metadata toMetadata(PDDocumentInformation documentInformation) {
```

Comment:
I am wondering if we should make inclusion of metadata configurable and don't include all metadata fields by default.
It might contain a lot of useless information and will just pollute the storage.
WDYT?

Reply:
Maybe add a method to DocumentParser which has a second parameter `Set<String> metadataKeys` to be retrieved? What if the user wants everything because he doesn't know what is available?

Reply:
With new method it would be complicated because you would need to modify all parsers and loaders.
What about adding a ctor `ApachePdfBoxDocumentParser(boolean includeMetadata)` instead?
User can filter out needed keys after parsing. But by default `includeMetadata` should be `false` to avoid pollution.